docs/user-guide/knowledge-distillation.rst

```diff
@@ -0,0 +1,362 @@
+            && echo "---------------" \
+            && echo "Starting..." \
+            && cd ${GPFS} \
+            && export PYTHONPATH="${GPFS}:${PYTHONPATH}" \
+            && export HYDRA_FULL_ERROR=1 \
+            && python examples/nlp/synthetic_data_gen/compute_topk_logits.py \
+                  trainer.num_nodes=\${SLURM_JOB_NUM_NODES} \
+                  trainer.devices=\${SLURM_NTASKS_PER_NODE} \
+                  trainer.precision=bf16 \
+                  pretrained_checkpoint.restore_from_path=teacher_checkpoint/Nemotron-3-8B-Chat-4k-SFT.nemo \
+                  model.megatron_amp_O2=True \
+                  model.tensor_model_parallel_size=1 \
+                  data.chat=True \
+                  data.sample=True \
+                  data.num_workers=0 \
+                  data.data.max_seq_length=4096 \
+                  data.data.file_path=data/oasst/train.jsonl \
+                  data.data.add_eos=False \
+                  data.data.hf_dataset=True \
+                  top_k=4 \
+                  model.global_batch_size=16 \
+                  model.micro_batch_size=2 \
+                  start_from_idx=0 \
+                  end_at_idx=56439 \
+                  output_path=data/oasst/train_with_logits_0.jsonl
+            EOF
+
+            srun -o $OUTFILE -e $ERRFILE --container-image=$CONTAINER $MOUNTS bash -c "${cmd}"
+            set +x
+
+
+You can also generate the teacher logits for the validation dataset by replacing these lines
+
+.. code-block:: bash
+
+   data.data.file_path=data/oasst/train.jsonl \
+   end_at_idx=56399 \
+   output_path=data/oasst/train_with_logits_0.jsonl
+
+with
+
+.. code-block:: bash
+
+   data.data.file_path=data/oasst/val.jsonl \
+   end_at_idx=2937 \
+   output_path=data/oasst/val_with_logits_0.jsonl
+ 
+Note that storing the teacher's logits can be quite memory intensive. To avoid going out of memory when loading the data,
+the data is loaded into memory in chunks. The example above uses a single chunk. To use multiple chunks, run the code multiple times, changing the ``start_from_idx`` and ``end_at_idx`` indices to
+exhaust the entire dataset:
+
+.. code-block:: bash
+
+   start_from_idx=${START_FROM_IDX} \
+   end_at_idx=${END_AT_IDX} \
+   output_path=data/oasst/train_with_logits_${CHUNK_INDEX}.jsonl
+
+Each time the code is run, a single chunk will be produced. Note that the output path should be suffixed with the chunk index. The index is expected to range from 0 to num_chunks - 1.
+
+Step 4: Fine-tune the student
+#############################
+
+Once the data has been prepared, you are ready to fine-tune the student model.  This step takes around 50 minutes on 8 H100 80G GPUs.
+
+.. tab-set::
+
+    .. tab-item:: Terminal
+        :sync: key3
+
+         To run fune-tuning on the terminal directly, use the following command. For successful execution, ensure that the NeMo-Aligner repository is set as your current working directory.
```

Comment:
fix spelling.

To run fine-tuning on the terminal directly, use the following command. For successful execution, ensure that the NeMo-Aligner repository is set as your current working directory.